forc-plugins/forc-publish/src/forc_pub_client.rs

```diff
@@ -50,16 +52,49 @@ impl ForcPubClient {
             .header("Content-Type", "application/gzip")
             .body(file_bytes)
             .send()
-            .await?;
-
-        let status = response.status();
+            .await;
 
-        if status.is_success() {
-            // Extract `upload_id` from the response if available
-            let upload_response: UploadResponse = response.json().await?;
-            Ok(upload_response.upload_id)
+        if let Ok(response) = response {
+            let mut stream = response.bytes_stream();
+
+            // TODO: Close stream
+            while let Some(chunk) = stream.next().await {
+                match chunk {
+                    Ok(bytes) => {
+                        let event_str = String::from_utf8_lossy(&bytes);
+                        for event in event_str.split("\n\n") {
+                            if let Some(stripped) = event.strip_prefix("data:") {
+                                let data = &stripped.trim();
+                                if let Ok(upload_response) =
+                                    serde_json::from_str::<UploadResponse>(data)
+                                {
+                                    return Ok(upload_response.upload_id);
+                                } else if data.starts_with("{") {
+                                    // Attempt to parse error from JSON
+                                    return Err(Error::ApiResponseError {
+                                        status: StatusCode::INTERNAL_SERVER_ERROR,
+                                        error: data.to_string(),
+                                    });
+                                } else {
+                                    // Print the event data, replacing the previous message.
+                                    print!("\r\x1b[2K  =>  {}", data);
+                                    use std::io::{stdout, Write};
+                                    stdout().flush().unwrap();
+                                }
+                            } else if event.starts_with(":") {
+                                // Do nothing. These are keep-alive events.
+                            }
```

Comment:
```suggestion
                            }
```

Do we need to have this if we are not doing anything?

Reply:
you could comment this out - as for this this is a redundant check (and will unneecessarily use compute (although minor) if top case is not matched)